xdem/spatial_tools.py

```diff
@@ -98,65 +98,101 @@ def merge_bounding_boxes(bounds: list[rio.coords.BoundingBox], resolution: float
         for key in "bottom", "left":
             max_bounds[key] = np.nanmin([max_bounds[key], bound.__getattribute__(key)])
 
-    for key in max_bounds:
-        modulo = max_bounds[key] % resolution
-        max_bounds[key] -= modulo
-
-        if key in ["right", "top"] and modulo > 0:
-            max_bounds[key] += resolution
+    # Make sure that extent is a multiple of resolution
+    for key1, key2 in zip(("left", "bottom"), ("right", "top")):
+        modulo = (max_bounds[key2] - max_bounds[key1]) % resolution
+        max_bounds[key2] += modulo
 
     return rio.coords.BoundingBox(**max_bounds)
 
 
-def merge_rasters(rasters: list[gu.georaster.Raster], reference: int = 0, merge_algorithm: Callable = np.nanmean,
-                  resampling_method: Union[str, rio.warp.Resampling] = "nearest") -> gu.georaster.Raster:
+def merge_rasters(rasters: list[gu.georaster.Raster], reference: int = 0,
```

Comment:
I think the functionality with using a non-included reference is great. I do believe that the syntax can be a bit more clear, however. 

Right now, the list one gives would be `[ref (which will not be included), nonref0, nonref1, nonref2 ... ]` which I think will take a while for people to get. What about having the `reference` argument either take a list index (like it does currently) or take a separate raster? This way, all given rasters in the `rasters` list will always be merged. This would make `include_ref` redundant but `use_ref_bounds` is still great to have. The function signature would then be:

```python
def merge_rasters(rasters: list[gu.Raster], reference: Union[int, gu.Raster] = 0, merge_algorithm, Union[Callable, list[Callable]] = np.nanmean, resampling_method: Union[str, rio.warp.Resampling] = "bilinear", use_ref_bounds: bool = False) -> gu.Raster:
```
What do you think about that?

Reply:
Sounds good, I'll implement that!

Reply:
Added in 2a40981